packages/hash/src/index.ts

```diff
@@ -2,62 +2,69 @@
 // Inspired by https://github.com/garycourt/murmurhash-js
 // Ported from https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash2.cpp#L37-L86
 
+const encoder = new TextEncoder()
```

Comment:
CI failure, `TextEncoder` doesn't exist in your testing environment, not sure where you prefer to fix this (here or adding it to the testing env).

Reply:
Fixing it in the test environment is preferred *if* we are sure this API will be available in all reasonable target environments. Note that, IIRC, we technically still support IE11. 😢

Reply:
That's terrible :| It's the most substantial part of the improvement.

I could remove it, or I could split the implementation in two functions based on `typeof TextEncoder === 'undefined'`, but that's more bundle size and if I have a choice, I'd rather put those bytes towards adding the wasm implementation.

Reply:
Actually I could also write an encoder specifically for IE11, based on `.charCodeAt()`, I think I could fit that in a 100 bytes.

Reply:
> Actually I could also write an encoder specifically for IE11, based on .charCodeAt(), I think I could fit that in a 100 bytes.

yeah, the good thing is that we don't need it to be fully spec-compliant - we only need to support the things we care about

I'm open to different solutions to this, even at the expense of some added bundlesize (but well, we also cant add too much, right? :P). It's just a concern that we have right now. The next version will be free to use TextEncoder directly.

Reply:
I have added a legacy encoder implementation for IE11. The benchmarks are unchanged for modern engines. I tested it manually by setting `hasTextEncoder` to `false`, but I don't know how to test it in IE11.

This is ready for review as far as I can tell.